src/java.desktop/share/classes/javax/swing/JSplitPane.java

```diff
@@ -371,24 +371,30 @@ public JSplitPane(int newOrientation,
     public void setComponentOrientation(ComponentOrientation orientation) {
         ComponentOrientation curOrn = this.getComponentOrientation();
         super.setComponentOrientation(orientation);
+        Component comp = null;
```

Comment:
Rename comp to tmp? Or swap?
And also reduce its scope by moving the declaration where it is assigned, this makes the code clearer. 
Maybe use var even?